R/src_dbi.R

```diff
@@ -85,6 +85,16 @@ tbl.src_dbi <- function(src, from, ...) {
   tbl_sql(c(subclass, "dbi"), src = src, from = from, ...)
 }
 
+# Internal calls to `tbl()` should be avoided in favor of tbl_src_dbi().
+# The former may query the database for column names if `vars` is omitted,
+# the latter always requires `vars`.
+tbl_src_dbi <- function(src, from, vars) {
```

Comment:
How about just adding something like:

```R
tbl <- function(...) {
  abort("Use `tbl_src_dbi()` instead")
}
```

I forgot that we were already automatically pulling in the variable names if they weren't supplied.

Reply:
Unfortunately this doesn't work because we are importing `tbl()` from dplyr and also declaring methods for it. As soon as we add the override, the dbplyr vignette fails rendering because the `tbl.sqlite()` method can't be found. I tried various hacks, including very weird ones.

We still could inspect our source code in a test.

Reply:
Ok, let's just leave it as is then.